src/core/error_helpers.js

```diff
@@ -265,28 +265,111 @@ function friendlyWelcome() {
   report(str, 'println', '#4DB200'); // auto dark green
 } */
 
-// This is a list of p5 functions/variables that are commonly misused
-// by beginners at top-level code, outside of setup/draw. We'd like to
-// detect these errors and help the user by suggesting they move them
+// This is a lazily-defined list of p5 symbols that may be
+// misused by beginners at top-level code, outside of setup/draw. We'd like
+// to detect these errors and help the user by suggesting they move them
 // into setup/draw.
 //
 // For more details, see https://github.com/processing/p5.js/issues/1121.
-var misusedAtTopLevelCode = [
-  'color',
-  'random'
-];
+var misusedAtTopLevelCode = null;
+var FAQ_URL = 'https://github.com/processing/p5.js/wiki/' +
+              'Frequently-Asked-Questions' +
+              '#why-cant-i-assign-variables-using-p5-functions-and-' +
+              'variables-before-setup';
 
-function helpForMisusedAtTopLevelCode(e) {
-  misusedAtTopLevelCode.forEach(function(name) {
-    if (e.message && e.message.indexOf(name) !== -1) {
-      console.log('%c Did you just try to use p5.js\'s \'' + name + '\' ' +
-                  'function or variable? If so, you may want to ' +
-                  'move it into your sketch\'s setup() function.',
-                  'color: #B40033' /* Dark magenta */);
+function defineMisusedAtTopLevelCode() {
+  var uniqueNamesFound = {};
+
+  var getSymbols = function(obj) {
+    return Object.getOwnPropertyNames(obj).filter(function(name) {
+      if (name[0] === '_') {
+        return false;
+      }
+      if (name in uniqueNamesFound) {
+        return false;
+      }
+
+      uniqueNamesFound[name] = true;
+
+      return true;
+    }).map(function(name) {
+      var type;
+
+      if (typeof(obj[name]) === 'function') {
+        type = 'function';
+      } else if (name === name.toUpperCase()) {
+        type = 'constant';
+      } else {
+        type = 'variable';
+      }
+
+      return {name: name, type: type};
+    });
+  };
+
+  misusedAtTopLevelCode = [].concat(
+    getSymbols(p5.prototype),
+    // At present, p5 only adds its constants to p5.prototype during
+    // construction, which may not have happened at the time a
+    // ReferenceError is thrown, so we'll manually add them to our list.
+    getSymbols(require('./constants'))
```

Comment:
I was actually kind of surprised that the constants like `PI` aren't added to `p5.prototype` until `p5`'s constructor. Was this intentional?

It's particularly odd to me because [the comment in `core.js`](https://github.com/processing/p5.js/blob/f1335ba8e65ea391712ff060ffe5c7ee850ba9b9/src/core/core.js#L443-L446) says `// attach constants to p5 instance`, yet the code is actually adding the constants to `p5.prototype`, which is part of the class, rather than the instance. I could file a separate bug for this if needed.